test/cql-pytest/test_filtering.py

```diff
@@ -239,6 +241,22 @@ def test_filtering_with_subscript(cql, test_keyspace, cassandra_bug):
         with pytest.raises(InvalidRequest, match='Unsupported unset map key for column m1'):
             cql.execute(stmt, [UNSET_VALUE])
 
+        # check subscripted list filtering
+        cql.execute(f"INSERT INTO {table} (p, l) VALUES (11, ['zero', 'one', 'two'])")
+        res = cql.execute(f"SELECT p FROM {table} WHERE l[1] = 'one' ALLOW FILTERING")
+        assert list(res) == [(11,)]
+        res = cql.execute(f"SELECT p FROM {table} WHERE l[1] = 'eight' ALLOW FILTERING")
+        assert list(res) == []
+        res = cql.execute(f"SELECT p FROM {table} WHERE l[-5] = 'minus five' ALLOW FILTERING")
+        assert list(res) == []
+        res = cql.execute(f"SELECT p FROM {table} WHERE l[3] = 'three' ALLOW FILTERING")
+        assert list(res) == []
+
+        # check that NULL lists don't confuse the system
+        cql.execute(f"INSERT INTO {table} (p) VALUES (12)")
+        res = cql.execute(f"SELECT p FROM {table} WHERE l[1] = 'one' ALLOW FILTERING")
+        assert list(res) == [(11,)]
```

Comment:
Good you tested that. Would also be nice to check that a null parameter, l[1] = null, does not cause a mess.
Also, please add checks for error conditions - e.g., `l[3] = 3` should complain that 3 is the wrong type, and so should `l['hello'] = 'hi'`.